### PR TITLE
Migrate: Gemma3nAudioConverter from TF ops to keras.ops

### LINKED ISSUE #2951: [PyGrain Migration] 7.  Migrate AudioConverter(Whisper, Moonshine, Gemma3n, Gemma4)
- Port each converter's ops (framing, STFT, mel filterbank, log scaling) to keras.ops under a CPU device scope, or NumPy where simpler  - follow whatever the ImageConverter port used for consistency.
- Verify numerics

### PR DESCRIPTION
**Pull request by maitry63**
This PR migrates Gemma3nAudioConverter from TensorFlow ops to keras.ops.

Replaces TensorFlow-specific audio operations with Keras equivalents, including 
- tf.signal.frame → keras.ops.extract_sequences 
-  tf.signal.rfft → keras.ops.rfft, 
while preserving batched/unbatched audio handling, spectrogram padding, and attention-mask behavior.

Fixes: #2951 (1st part)

Numerical verification [colab](https://colab.sandbox.google.com/drive/1c1p6lQ1VpdLOTmo_UBeNIyUC8m-aANbI?resourcekey=0-RXQw0HI3WcFkDh0cCzUruw#scrollTo=NzenbbCpRl3Y).

### Numerical Verification Results

Verified numerical equivalence against pure NumPy reference :

  - Test 1 (Sine Wave): Log-Mel max diff = 9.64e-03 | Linear-Mel max diff = 1.14e-04 [PASS]
  - Test 2 (Normalized Audio): max diff = 2.19e-05 [PASS]
  - Test 3 (Silent Audio Floor): max diff = 3.17e-07 [PASS]
  - Test 4 (Batch Invariance): max diff = 0.00e+00 [PASS]

## Checklist

- [x] I have added all the necessary unit tests for my change.
- [x] I have verified that my change does not break existing code and works with all backends (TensorFlow, JAX, and PyTorch).
- [x] My PR is based on the latest changes of the main branch (if unsure, rebase the code).
- [x] I have followed the Keras Hub [Model contribution guidelines](https://github.com/keras-team/keras-hub/blob/master/CONTRIBUTING_MODELS.md) in making these changes.
- [x] I have followed the Keras Hub [API design guidelines](https://github.com/keras-team/keras-hub/blob/master/API_DESIGN_GUIDE.md) in making these changes.
- [x] I have signed the [Contributor License Agreement](https://cla.developers.google.com/about).

### CODE REVIEW

**Comment by laxmareddyp** [comment]
closing this in favor of #3064
